src/components/general/Dropdown/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, {
+    useState,
+    useRef,
+    useCallback,
+    useEffect,
+    MutableRefObject,
+    ReactNode,
+    FC,
+} from 'react';
+import {
+    TextInput,
+    DropdownArrow,
+    Menu,
+    useClickOutsideOverlayPortal,
+    useOverlayContainer,
+    useRelativePositioning,
+} from '@equinor/fusion-components';
+import styles from './styles.less';
+import { useFusionContext, FusionContext } from '@equinor/fusion';
+import { createPortal } from 'react-dom';
+import { Router } from 'react-router';
+
+type DropdownSelections = {
+    title: string;
+    key: string;
+};
+
+type DropdownProps = {
+    label?: string;
+    selections: DropdownSelections[];
+    onSelect?: (item: DropdownSelections) => void;
+    selected?: string;
+};
+
+const Dropdown: FC<DropdownProps> = ({ selections, label, onSelect, selected }) => {
+    const [open, setOpen] = useState(false);
+    const [inputValue, setInputValue] = useState('');
+    const [dropdownSelections, setDropdownSelections] = useState(selections);
+
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const inputContainerRef = useRef<HTMLDivElement | null>(null);
+
+    const close = useCallback(() => open && setOpen(false), [open]);
+
+    const filterSearch = useCallback(inputValue => {
+        setDropdownSelections(
+            selections.filter(selection =>
+                selection.title.toLowerCase().includes(inputValue.toLowerCase())
+            )
+        );
+    }, []);
```

Comment:
Should probably be a dependency on the selections/options prop here to rebuild the callback when they change